fx-src/FirefoxStorage.js

```diff
@@ -108,12 +108,16 @@ const currentSchemaVersion = 1;
 export default class FirefoxAdapter extends BaseAdapter {
   constructor(collection, options={}) {
     super();
+    const {sqliteHandle=null} = options;
     this.collection = collection;
-    this._connection = null;
+    this._connection = sqliteHandle;
     this._options = options;
   }
 
-  _init(connection) {
+  // We need to be capable of calling this from "outside" the adapter
+  // so that someone can initialize a connection and pass it to us in
+  // adapterOptions.
+  static _init(connection) {
```

Comment:
Yeah, I was hesitant to even try fighting the build system. When @leplatrem was in New York, I saw him struggling with #562 and how there was only allowed to be one exported symbol. I was in a hurry at the time I was writing this, and I knew that #562 would probably follow this -- when it does, all of this code becomes part of the Gecko codebase. Because of that, I was comfortable considering `_init` as not part of the public API from the perspective of Gecko.